File: backend/routing/logging/clear.go
package routes_logging

import (
	"aura/logging"
	"aura/utils"
	"aura/utils/httpx"
	"fmt"
	"net/http"
	"os"
	"path"
	"strings"
)

func ClearLogs(w http.ResponseWriter, r *http.Request) {
	ctx, ld := logging.CreateLoggingContext(r.Context(), r.URL.Path)
	logAction := ld.AddAction("Clear Logs", logging.LevelInfo)
	ctx = logging.WithCurrentAction(ctx, logAction)

	// Get the query parameter to determine which logs to clear (current file or old files)
	clearOption := r.URL.Query().Get("option") // "current" or "old"

	if clearOption == "" {
		clearOption = "current" // Default to clearing current log file
	} else if clearOption != "current" && clearOption != "old" {
		logAction.SetError("Invalid clear option. Must be 'current' or 'old'.", "", nil)
		httpx.SendResponse(w, ld, nil)
		return
	}

	// Check if the logs folder exists
	Err := utils.CreateFolderIfNotExists(ctx, logging.LogFolder)
	if Err.Message != "" {
		httpx.SendResponse(w, ld, nil)
	}

	// Clear logs based on the option
	var clearMessage string
	switch clearOption {
	case "current":
		// If the file exists, truncate it to clear its content
		if _, err := os.Stat(logging.LogFilePath); err == nil {
			f, err := os.OpenFile(logging.LogFilePath, os.O_TRUNC|os.O_WRONLY, 0644)
			if err != nil {
				logAction.SetError("Failed to open log file for truncation", "Ensure the file is not read-only and you have permissions", map[string]any{"error": err.Error()})
				httpx.SendResponse(w, ld, nil)
				return
			}
			// Ensure changes are flushed to disk
			if err := f.Sync(); err != nil {
				logAction.SetError("Failed to sync log file after truncation", "Ensure the file system is not read-only", map[string]any{"error": err.Error()})
				f.Close()
				httpx.SendResponse(w, ld, nil)
				return
			}
			f.Close()
			clearMessage = "Current log file cleared successfully"
		} else {
			ld.Status = logging.StatusWarn
			clearMessage = "No current log file to clear"
		}
	case "old":
		// Delete all log files except the current one
		files, err := os.ReadDir(logging.LogFolder)
		if err != nil {
			logAction.SetError("Failed to read log directory", "Ensure the logs folder exists and is accessible", map[string]any{"error": err.Error()})
			httpx.SendResponse(w, ld, nil)
			return
		}

		deletedFiles := 0
		for _, file := range files {
			if strings.HasSuffix(file.Name(), ".log") && file.Name() != "aura.log" {
				err := os.Remove(path.Join(logging.LogFolder, file.Name()))
				if err != nil {
					logAction.SetError(fmt.Sprintf("Failed to delete log file: %s", file.Name()), "Ensure the file is not in use and you have permissions", map[string]any{"error": err.Error()})
					httpx.SendResponse(w, ld, nil)
					return
				}
				deletedFiles++
			}
		}
		if deletedFiles == 0 {
			ld.Status = logging.StatusWarn
			clearMessage = "No old log files to clear"
		} else {
			clearMessage = fmt.Sprintf("Cleared %d old log file(s) successfully", deletedFiles)
		}
	}

	httpx.SendResponse(w, ld, clearMessage)
}

File: frontend/src/services/logs/clear.ts
import apiClient from "@/services/api-client";
import { ReturnErrorMessage } from "@/services/api-error-return";

import { log } from "@/lib/logger";

import { APIResponse } from "@/types/api/api-response";

export interface ClearLogFiles_Response {
  message: string;
}

export const ClearLogFiles = async (clearCurrent: boolean = false): Promise<APIResponse<ClearLogFiles_Response>> => {
  log("INFO", "API - Logs", "Clear Old Logs", `Clearing old logs, clearCurrent=${clearCurrent}`);
  try {
    const params = { option: clearCurrent ? "current" : "old" };
    const response = await apiClient.delete<APIResponse<ClearLogFiles_Response>>(`/logs`, { params });
    if (response.data.status === "error") {
      throw new Error(response.data.error?.message || "Unknown error clearing old logs");
    } else {
      log("INFO", "API - Logs", "Clear Old Logs", "Cleared old logs successfully", response.data);
    }
    return response.data;
  } catch (error) {
    log(
      "ERROR",
      "API - Logs",
      "Clear Old Logs",
      `Failed to clear old logs: ${error instanceof Error ? error.message : "Unknown error"}`,
      error
    );
    return ReturnErrorMessage<ClearLogFiles_Response>(error);
  }
};
